internal/pkg/agent/application/upgrade/artifact/artifact_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License 2.0;
+// you may not use this file except in compliance with the Elastic License 2.0.
+
+package artifact
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	agtversion "github.com/elastic/elastic-agent/pkg/version"
+)
+
+func TestGetArtifactName(t *testing.T) {
+	version, err := agtversion.ParseVersion("9.1.0")
+	require.NoError(t, err)
+
+	tests := map[string]struct {
+		a               Artifact
+		version         agtversion.ParsedSemVer
+		operatingSystem string
+		arch            string
+		expectedName    string
+		expectedErr     string
```

Comment:
Nit: not used anymore in these tests, also `version`, `operatingSystem` and `arch` do not differ in the below testcases, so might be able to tidy this up a bit.